src/coreclr/tools/Common/Compiler/Win32Resources/ResourceData.cs

```diff
@@ -67,6 +89,44 @@ public byte[] FindResource(ushort name, ushort type, ushort language)
             return FindResourceInternal(name, type, language);
         }
 
+        /// <summary>
+        /// Add or update resource
+        /// </summary>
+        public void AddResource(string name, string type, ushort language, byte[] data) => AddResourceInternal(name, type, language, data);
+
+        /// <summary>
+        /// Add or update resource
+        /// </summary>
+        public void AddResource(ushort name, string type, ushort language, byte[] data) => AddResourceInternal(name, type, language, data);
+
+        /// <summary>
+        /// Add or update resource
+        /// </summary>
+        public void AddResource(string name, ushort type, ushort language, byte[] data) => AddResourceInternal(name, type, language, data);
+
+        /// <summary>
+        /// Add or update resource
+        /// </summary>
+        public void AddResource(ushort name, ushort type, ushort language, byte[] data) => AddResourceInternal(name, type, language, data);
+
+        public IEnumerable<(object name, object type, ushort language, byte[] data)> GetAllResources()
+        {
+            return _resTypeHeadID.SelectMany(typeIdPair => CopyResType(typeIdPair.Key, typeIdPair.Value))
+                .Concat(_resTypeHeadName.SelectMany(typeNamePair => CopyResType(typeNamePair.Key, typeNamePair.Value)));
+
+            IEnumerable<(object name, object type, ushort language, byte[] data)> CopyResType(object type, ResType resType)
+            {
+                return resType.NameHeadID.SelectMany(nameIdPair => CopyResName(type, nameIdPair.Key, nameIdPair.Value))
+                    .Concat(resType.NameHeadName.SelectMany(nameNamePair =>
+                        CopyResName(type, nameNamePair.Key, nameNamePair.Value)));
+            }
+
+            IEnumerable<(object name, object type, ushort language, byte[] data)> CopyResName(object type, object name, ResName resType)
+            {
+                return resType.Languages.Select((lang) => (name, type, lang.Key, lang.Value.DataEntry));
```

Comment:
Can you switch this to return `type, name` instead of `name, type`? I find it confusing that the order is flipped from `AddResourceInternal`.

Reply:
Umm... Old public function in `ResourceData`,`FindResource`, uses `name, type, language` order so I think it's better to change signature of `AddResource` and `AddResourceInternal`, and keep `name, type, language` order in GetAllResources.